webpay/pay/tests/test_tasks.py

```diff
@@ -577,6 +577,46 @@ def test_bango_used_when_boku_supported_but_not_setup_in_solitude(self):
             seller_product__external_id='external_id'
         )
 
+    def test_bango_used_when_boku_supported_but_no_price(self):
+        mcc, mnc = api.BokuProvider.network_data.keys()[0]
+        self.providers = api.ProviderHelper.supported_providers(
+            mcc=mcc, mnc=mnc)
+
+        self.solitude.generic.seller.get_object_or_404.return_value = {
+            'resource_pk': 1,
+            'uuid': self.boku_seller_uuid,
+            'resource_uri': '/generic/seller/1/'
+        }
+
+        self.solitude.generic.product.get_object_or_404.return_value = {
+            'resource_uri': '/generic/product/1/',
+            'public_id': 'public_id',
+            'seller_uuids': {
+                'bango': self.bango_seller_uuid,
+                'boku': self.boku_seller_uuid,
+            },
+            'seller': '/generic/seller/1',
+            'external_id': 'external_id',
+        }
+
+        def no_boku_price(pricePoint, provider):
+            if provider == 'boku':
+                return {'prices': []}
+            return {'prices': [{'price': '0.99'}]}
+
+        pricey = mock.Mock()
+        pricey.get_object.side_effect = no_boku_price
+        self.mkt.webpay.prices.return_value = pricey
+
+        tasks.start_pay(self.transaction_uuid, self.notes, self.user_uuid,
+                        [p.name for p in self.providers])
+
+        self.solitude.bango.product.get_object_or_404.assert_called_with(
+            seller_product__seller=1,
+            seller_product__external_id='external_id'
+        )
+        ok_(self.mkt.webpay.prices.call_count, 2)
```

Comment:
could this also assert a call to ``self.solitude.bango.billing.post`` ? That seems like the important one.